agent/client/workflow_listen_for_otlp_connection_tests.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/kubeshop/tracetest/agent/proto"
+)
+
+func (c *Client) startOTLPConnectionTestListener(ctx context.Context) error {
+	client := proto.NewOrchestratorClient(c.conn)
+
+	stream, err := client.RegisterOTLPConnectionTestListener(ctx, c.sessionConfig.AgentIdentification)
+	if err != nil {
+		return fmt.Errorf("could not open agent stream: %w", err)
+	}
+
+	go func() {
+		for {
+			req := proto.OTLPConnectionTestRequest{}
+			err := stream.RecvMsg(&req)
+			if isEndOfFileError(err) || isCancelledError(err) {
+				return
+			}
+
+			reconnected, err := c.handleDisconnectionError(err)
+			if reconnected {
+				return
+			}
+
+			if err != nil {
+				log.Println("could not get message from otlp connection stream: %w", err)
+				time.Sleep(1 * time.Second)
+				continue
+			}
+
+			// TODO: Get ctx from request
```

Comment:
Philosophical question: is there a way to get the context from the request?
My impression is that there is no way to get these contexts due to the passive nature of getting the messages.

Could we create a context in the future and change it by monitoring the stream connection?

Reply:
We would need to propagate headers through the messages to be able to create a context from the request (propagate the otel headers)